Exercises/NoModules/Chapter 03/Soln3_01.cpp

```diff
@@ -0,0 +1,30 @@
+// Exercise 3-1. Output integer and its 1's complement in decimal and hexadecimal.
+
+// This tests how well you remember string formatting using std::format()
+// (see Chapter 2 if you forgot some of the formatting options), 
+// as well as two�s complement binary encoding and bitwise ~.
+
+#include <iostream>
+#include <format>
+
+// See Appendix A (available online) for static_assert().
+// You should adjust the field widths of the table if ints have a different size.
+static_assert(sizeof(int) == 4, "This program assumes 32 bit ints");
```

Comment:
why?
you are using sizeof(int), so I assume the reader knows about it.
Then you can just have (sizeof(int)*8+2) for the field width instead of hard-coding to 34 and remove this static_assert.

Reply:
The reader knows sizeof, but not dynamic field lengths yet. That is covered in Chapter 7. So hard-coding it is!